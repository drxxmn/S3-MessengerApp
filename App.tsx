import { StatusBar } from 'expo-status-bar';
import React, {useEffect} from "react";
import { SafeAreaProvider } from 'react-native-safe-area-context';

import useCachedResources from './hooks/useCachedResources';
import useColorScheme from './hooks/useColorScheme';
import Navigation from './navigation';

import {
    Auth,
    API,
    graphqlOperation,
} from 'aws-amplify'
import { getUser } from './src/graphql/queries'
import { createUser } from './src/graphql/mutations'

import { withAuthenticator } from 'aws-amplify-react-native'

import { Amplify } from 'aws-amplify'
import awsconfig from './src/aws-exports'
Amplify.configure(awsconfig)

const randomImages = [
    'https://hieumobile.com/wp-content/uploads/avatar-among-us-2.jpg',
    'https://hieumobile.com/wp-content/uploads/avatar-among-us-3.jpg',
    'https://hieumobile.com/wp-content/uploads/avatar-among-us-6.jpg',
    'https://hieumobile.com/wp-content/uploads/avatar-among-us-9.jpg',
]

function App() {
  const isLoadingComplete = useCachedResources();
  const colorScheme = useColorScheme();

  const getRandomImage = () => {
      return randomImages[Math.floor(Math.random()* randomImages.length)]
  }

  //run this snipped only when the App is first mounted
  useEffect(() => {
    const fetchUser = async () => {
      //get authenticated user from auth
      const userInfo = await Auth.currentAuthenticatedUser({bypassCache: true});

      if (userInfo) {
        //get the user from backend with user SUB from auth
        const userData = await API.graphql(
            graphqlOperation(
                getUser,
                { id: userInfo.attributes.sub }
            )
        )
          if(userData.data.getUser) {
              console.log("user is already registered in database");
              return;
          }

          const newUser = {
              id: userInfo.attributes.sub,
              name: userInfo.username,
              imageUri: getRandomImage(),
              status: "Hey, I'm using Lune Messenger!"
          }

          await API.graphql(
              graphqlOperation(
                createUser,
                  { input: newUser }
              )
          )
        //if there is no user in DB with the Id, then create one
      }

    }
    fetchUser();
    }, [])

  if (!isLoadingComplete) {
    return null;
  } else {
    return (
      <SafeAreaProvider>
        <Navigation colorScheme={colorScheme} />
        <StatusBar />
      </SafeAreaProvider>
    );
  }
}

export default withAuthenticator(App)
