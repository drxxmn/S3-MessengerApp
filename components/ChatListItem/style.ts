import {Appearance, StyleSheet, useColorScheme} from 'react-native';
import Colors from "../../constants/Colors";

const colorScheme = Appearance.getColorScheme();
const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        width: "100%",
        flex: 1,
        justifyContent: 'space-between',
        padding: 10,
    },
    leftContainer: {
        flexDirection: 'row',
    },
    midContainer: {
        justifyContent: 'space-around',
    },
    avatar: {
        width: 60,
        height: 60,
        marginRight: 15,
        borderRadius: 50,
    },
    username: {
        fontWeight: 'bold',
        fontSize: 18,
        color: Colors[colorScheme].text,
    },
    lastMessage: {
        fontSize: 16,
        color: 'grey',
        
    },
    time: {
        fontSize: 14,
        color: 'grey',
    },
});

export default styles;