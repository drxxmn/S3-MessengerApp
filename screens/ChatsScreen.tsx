import {FlatList, StyleSheet} from 'react-native';
import * as React from 'react';
import EditScreenInfo from '../components/EditScreenInfo';
import { Text, View } from '../components/Themed';
import { RootTabScreenProps } from '../types';
import ChatListItem from "../components/ChatListItem";
import {
    API,
    graphqlOperation,
    Auth,
} from "aws-amplify";

import chatRooms from '../data/ChatRooms';
import NewMessageButton from "../components/NewMessageButton";
import {useEffect, useState} from "react";

import { getUser } from './queries';

export default function ChatsScreen() {

    const [chatRooms, setChatRooms] = useState([]);

  useEffect(() => {
    const fetchChatRooms = async () => {
      try {
        const userInfo = await Auth.currentAuthenticatedUser();

        const userData = await API.graphql(
            graphqlOperation(
                getUser, {
                  id: userInfo.attributes.sub,
                }
            )
        )
          setChatRooms(userData.data.getUser.chatRoomUser.items);
          console.log(userData.data.getUser.chatRoomUser.items);

      } catch (e) {
        console.log(e)
      }
    }
    fetchChatRooms();
  }, []);

  return (
    <View style={styles.container}>
      <FlatList
          style={{width: '100%'}}
          data={chatRooms}
          renderItem={({item}) => <ChatListItem chatRoom={item.chatRoom} />}
          keyExtractor={(item) => item.id}
      />
        <NewMessageButton />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },

});
