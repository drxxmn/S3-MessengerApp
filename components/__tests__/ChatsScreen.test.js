import React from "react";
import renderer from 'react-test-renderer';
import 'react-native';
import newMessageButton  from "../NewMessageButton";
import ChatListItem from "../ChatListItem";
import ChatsScreen from "../../screens/ChatsScreen";
import App from "../../App";
import { render, fireEvent } from "@testing-library/react-native";

describe('test true is true', () => {
    it('true is true', () => {
        expect(true).toBe(true);
    });
}
);

it(`renders new message button correctly`, () => {
    const tree = renderer.create(newMessageButton).toJSON();

    expect(tree).toMatchSnapshot();
});

/*describe('chats screen',() => {
    it('should go to chat page',() => {
            const page = render(<ChatListItem />);
            const chatButton = page.getByTestId("chatRoomButton");
    })
})*/
