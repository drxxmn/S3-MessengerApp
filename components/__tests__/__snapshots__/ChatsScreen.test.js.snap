// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders new message button correctly 1`] = `null`;
